cfme/tests/configure/test_ntp_server.py

```diff
@@ -0,0 +1,83 @@
+from cfme import login
+from cfme.configure import configuration
+from cfme.fixtures import pytest_selenium as sel
+from datetime import datetime
+from fixtures.pytest_store import store
+from utils.conf import cfme_data
+from utils.log import logger
+from utils.randomness import generate_random_string
+from utils.ssh import SSHClient
+from utils.wait import wait_for
+
+import pytest
+
+
+def ssh_command_execution(command):
+    with SSHClient() as ssh:
+        status, msg = ssh.run_command("%s" % command)
+        if status == 0:
+            return msg
+
+
+def appliance_date():
+    return datetime.strptime(ssh_command_execution("date --iso-8601=hours").rsplit('-', 1)[0],
+        '%Y-%m-%dT%H')
+
+
+def test_ntp_crud():
+    """ Insert, Update and Delete the NTP servers """
+    # set from yaml file
+    configuration.configure_ntp_servers()
+
+    # set from random values
+    configuration.configure_ntp_servers(*(generate_random_string() for _ in range(3)))
+
+    configuration.unset_ntp_servers()
+
+
+def test_ntp_server_max_character():
+    configuration.configure_ntp_servers(*(generate_random_string(size=255) for _ in range(3)))
+
+
+def test_ntp_conf_file_update_check():
+    configuration.configure_ntp_servers()
+    for clock in cfme_data['clock_servers']:
+        assert ssh_command_execution("cat /etc/ntp.conf | grep %s" % clock).rsplit()
+
+
+def test_ntp_server_check():
+    orig_date = appliance_date()
+    past_date = ssh_command_execution("date --date='10 days ago' +%y%m%d")
+    ssh_command_execution("date --set=\"$(date +'%s %%H:%%M')\"" % past_date)
+    prev_date = appliance_date()
+    if prev_date != orig_date:
+        logger.info("Successfully modified the date in the appliance")
+        # Configuring the ntp server and restarting the appliance
+        yaml_config = store.current_appliance.get_yaml_config("vmdb")
+        ntp = yaml_config.get("ntp", None)
+        if not ntp:
+            yaml_config["ntp"] = {}
+        yaml_config["ntp"]["interval"] = '1.minute'
```

Comment:
Just a check. I saw in the configuration something that referenced, ```1.minutes``` I'm assuming ```1.minute``` is allowed as an abbreviation to ```1.minutes```?